api/core/app/apps/advanced_chat/app_runner.py

```diff
@@ -61,8 +76,7 @@ def run(self) -> None:
         if not app_record:
             raise ValueError("App not found")
 
-        workflow = self.get_workflow(app_model=app_record, workflow_id=app_config.workflow_id)
-        if not workflow:
+        if not self._workflow:
```

Comment:
The `if not self._workflow:` check appears unnecessary since the `workflow` parameter is required by the `AdvancedChatAppRunner` constructor.

If the intention is to ensure `self._workflow` is not `None`, I suggest moving this validation to the constructor with an assertion or explicit check for better debugging experience.